### ISSUE TITLE
Make multipart more flexible with file types

### ISSUE THREAD

**Issue by vincentjames501** [gnarroway/hato#13]
I need to be able to set a file name and content-type/mime-type of a file part without actually having a file written to disk (i.e. the source is an InputStream/byte array).

Currently hato.multipart doesn't allow this without me writing the file to disk.

Changing the following works for me:

```clojure
(defn- content-type
  [{:keys [content content-type]}]
  (cond
    content-type (str "Content-Type: " content-type)
    (string? content) "Content-Type: text/plain; charset=UTF-8"
    (instance? File content) (str "Content-Type: " (or (Files/probeContentType (.toPath content))
                                                       "application/octet-stream"))
    :else "Content-Type: application/octet-stream"))

(defn- content-disposition
  [{:keys [part-name name content file-name]}]
  (str "Content-Disposition: form-data; "
       (format "name=\"%s\"" (or part-name name))
       (when file-name (format "; filename=\"%s\"" file-name))
       (when (instance? File content) (format "; filename=\"%s\"" (.getName content)))))
```

I can then set `:multipart` to:

```
[{:name "file" :content byte-array :content-type "image/png" :file-name "image.png"}]
```

**Comment by gnarroway** [comment]
Hi, thank you again for using hato. The above seems reasonable. 

Would you be able to open a PR with tests?

Also the content-disposition may be a bit off if both file-name is specified and the content is a file.

**Comment by vincentjames501** [comment]
PR sent w/ tests and doc updates. Thanks @gnarroway !

**Comment by gnarroway** [comment]
Released in 0.6.0 thanks.
